src/metabase/util/fonts.clj

```diff
@@ -0,0 +1,45 @@
+(ns metabase.util.fonts
+  "font loading functionality."
+  (:require [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [metabase.util.files :as u.files]))
+
+(def ^:private font-path "./resources/frontend_client/app/fonts/")
+
+(defn- normalize-font-dirname
+  "Use a font's directory to derive a Display Name.
+
+  Split on dashes and take the first word, then change underscores to spaces.
+  This is done because of the 'Lato-v16-latin' directory name."
+  [dirname]
+  (-> dirname
+      (str/split #"-")
+      first
+      (str/replace #"_" " ")))
+
+(defn- contains-font-file?
+  [path]
+  ;; todo: expand this to allow other font formats?
+  (boolean (some #(str/includes? % ".woff") (u.files/files-seq path))))
+
+(defn- available-fonts*
+  []
+  (log/info (str "Reading available fonts from " font-path))
+  (->> font-path
+       u.files/get-path
+       u.files/files-seq
+       (filter contains-font-file?)
+       (map #(str/replace (str %) font-path ""))
+       (map normalize-font-dirname)
+       (sort-by #(str/lower-case %))))
```

Comment:
Consider using `(apply sorted-set-by str/lower-case)` here and `(contains? (available-fonts) font)` to implement `available-font?`.